src/views/CardStack.js

```diff
@@ -181,41 +164,24 @@ class CardStack extends Component {
     }
   }
 
-  _reset(resetToIndex: number, velocity: number): void {
+  _reset(resetToIndex: number): void {
     Animated.timing(this.props.position, {
         toValue: resetToIndex,
         duration: ANIMATION_DURATION,
         useNativeDriver: this.props.position.__isNative,
-        velocity: velocity * GESTURE_ANIMATED_VELOCITY_RATIO,
-        bounciness: 0,
       })
       .start();
   }
 
-  _goBack(backFromIndex: number, velocity: number) {
-    const {navigation, position, scenes} = this.props;
+  _goBack(backFromIndex: number) {
+    const {navigation, scenes} = this.props;
     const toValue = Math.max(backFromIndex - 1, 0);
-
-    // set temporary index for gesture handler to respect until the action is
-    // dispatched at the end of the transition.
-    this._immediateIndex = toValue;
-
-    Animated.timing(position, {
-        toValue,
-        duration: ANIMATION_DURATION,
-        useNativeDriver: position.__isNative,
-        velocity: velocity * GESTURE_ANIMATED_VELOCITY_RATIO,
-        bounciness: 0,
-      })
-      .start(({finished}) => {
-        this._immediateIndex = null;
-        const backFromScene = scenes.find(s => s.index === toValue + 1);
-        if (!this._isResponding && backFromScene) {
-          navigation.dispatch(
-            NavigationActions.back({ key: backFromScene.route.key })
-          );
-        }
-      });
+    const backFromScene = scenes.find(s => s.index === toValue + 1);
+    if (!this._isResponding && backFromScene) {
+      navigation.dispatch(
```

Comment:
I think the right way is using `Animated.spring` after gesture to achieve smooth transition, but we should take care the following screen transition animation, if we swipe back, the following screen transition should be done in 0ms, then the two issues would be resolved in the same time